src/rust/ide/src/ide.rs

```diff
@@ -81,14 +81,13 @@ impl IdeInitializer {
     /// Connect to language server.
     pub async fn open_project
     ( logger           : &Logger
-    , project_manager  : &impl project_manager::API
-    , project_metadata : &ProjectMetadata
+    , project_manager  : impl project_manager::API + 'static
```

Comment:
where this 'static comes from? I'm very careful about such constraints because they populate easily trough the codebase and its hard to prune them later. We need to understand exactly where such things come from. In case of functions that are boxed and stored somewhere its obvious, here its not and needs description

Reply:
I refactored this code to leave 'static out of it.